sourcegraph/sourcegraph
# Gate keep Code Insights app UI based on code insights API availability

## Problem 
Prior to this PR all code insights pages were either failing with page-level error about code insights has been disabled or had infinity spinners (since in some places we don't handler errors properly). As a quick fix for this in this PR we move code insights API check to enterprise package in order to reuse it to populate jscontext codeInsighsEnabled and based on this information render landing page for insight URLS if code insights api is disabled. 

It also extends a bit code insights checker helper (I added dotCom check there as well, since we don't support code insights on dot com yet)

## Test plan
- Run sg start enterprise 
- Check that you can't see any code insights app-like pages but just landing page
- Run sg start enterprise-codeinsights check that code insights app UI is available
